packages/webapp/src/containers/CropVarieties/useCropVarietyCatalogue.js

```diff
@@ -0,0 +1,262 @@
+
+  const _without_management_plan_ByCropId = _without_management_plan.filter(
+    (c) => c.crop_id === crop_id,
+  );
+
+  const managementPlansFilteredByLocations = useMemo(() => {
+    const locationFilter = cropCatalogueFilter[LOCATION];
+    const included = new Set();
+    for (const location_id in locationFilter) {
+      if (locationFilter[location_id].active) included.add(location_id);
+    }
+    if (included.size === 0) return managementPlansFilteredByFilterString;
+    return managementPlansFilteredByFilterString.filter((managementPlan) =>
+      included.has(managementPlan.location?.location_id),
+    );
+  }, [cropCatalogueFilter[LOCATION], managementPlansFilteredByFilterString]);
+
+  const managementPlansFilteredBySuppliers = useMemo(() => {
+    const supplierFilter = cropCatalogueFilter[SUPPLIERS];
+    const included = new Set();
+    for (const supplier in supplierFilter) {
+      if (supplierFilter[supplier].active) included.add(supplier);
+    }
+    if (included.size === 0) return managementPlansFilteredByLocations;
+    return managementPlansFilteredByLocations.filter((managementPlan) =>
+      included.has(managementPlan.supplier),
+    );
+  }, [cropCatalogueFilter[SUPPLIERS], managementPlansFilteredByLocations]);
+
+  const cropCatalogue = useMemo(() => {
+    const time = new Date(cropCatalogFilterDate).getTime();
+    const managementPlansByStatus = {
+      active: getCurrentManagementPlans(managementPlansFilteredBySuppliers, time),
+      planned: getPlannedManagementPlans(managementPlansFilteredBySuppliers, time),
+      past: getExpiredManagementPlans(managementPlansFilteredBySuppliers, time),
+    };
+    const managementPlansByCropId = {};
+    for (const status in managementPlansByStatus) {
+      for (const managementPlan of managementPlansByStatus[status]) {
+        if (!managementPlansByCropId.hasOwnProperty(managementPlan.crop_variety_id)) {
+          managementPlansByCropId[managementPlan.crop_variety_id] = {
+            active: [],
+            planned: [],
+            past: [],
+            crop_common_name: managementPlan.crop_common_name,
+            crop_translation_key: managementPlan.crop_translation_key,
+            imageKey: managementPlan.crop_translation_key?.toLowerCase(),
+            crop_id: managementPlan.crop_id,
+            crop_photo_url: managementPlan.crop_photo_url,
+            crop_variety_id: managementPlan.crop_variety_id,
+            crop_variety_name: managementPlan.crop_variety_name,
+          };
+        }
+
+        managementPlansByCropId[managementPlan.crop_variety_id][status].push(managementPlan);
+      }
+    }
+    const managementPlansByCropId_list = Object.values(managementPlansByCropId);
+    const managementPlansByCropIdWithNoPlans = managementPlansByCropId_list.reduce(
+      (previousValue, currentValue) => {
+        const no_plan_found = _without_management_plan_ByCropId.filter(
```

Comment:
Can `no_plan_found` be camel case for consistency?

Reply:
Updated with new changes. as asked, I am now using camel case for consistency.